Finmer.Editor/FormDocumentScene.cs

```diff
@@ -510,6 +510,49 @@ private void trvNodes_DragDrop(object sender, DragEventArgs e)
             Dirty = true;
         }
 
+        public TreeNode GetTreeNodeByKey(string key) 
+        {
+            //Prepare stack with root TreeNodes in it
+            Stack<TreeNode> stack = new Stack<TreeNode>();
+            foreach (TreeNode root_node in trvNodes.Nodes)
+                stack.Push(root_node);
+
+            // Flood-fill down the tree until the requested node is found
+            while (stack.Count > 0)
+            {
+                TreeNode node = stack.Pop();
+                var node_sn = (AssetScene.SceneNode)node.Tag;
+
+                // Links do not have keys
+                if (node_sn.IsLink)
+                    continue;
+
+                // Is this the node we're looking for?
+                if (node_sn.Key.Equals(key))
+                    return node;
+
+                // Recursively examine the node's children
+                foreach (TreeNode child in node.Nodes)
+                    stack.Push(child);
+            }
+
+            return null;
+        }
+
+        private void trvNodes_Link_DoubleClick(object sender, TreeNodeMouseClickEventArgs e) 
```

Comment:
When this event is triggered, the user may have double-clicked a non-link node, hence the name should not imply assumptions to the contrary.
```suggestion
        private void trvNodes_NodeMouseDoubleClick(object sender, TreeNodeMouseClickEventArgs e) 
```